breacharbiter_test.go

```diff
@@ -831,3 +925,597 @@ restartCheck:
+	chanPoint := alice.ChanPoint
+	breachTxn := forceCloseSummary.CloseTx
+	notifier.Spend(chanPoint, 100, breachTxn)
+
+	// Wait for the close observer to exit, all persistent effects should be
+	// observable after this point.
+	alice.WaitForClose()
+
+	// Since the handoff failed, the breach arbiter should not show the
+	// channel as breached, and the channel should also not have been marked
+	// pending closed.
+	assertNoArbiterBreach(t, brar, chanPoint)
+	assertNotPendingClosed(t, alice)
+
+	// Instantiate a second lightning channel for alice, using the state of
+	// her last channel.
+	aliceKeyPriv, _ := btcec.PrivKeyFromBytes(btcec.S256(),
+		alicesPrivKey)
+	aliceSigner := &mockSigner{aliceKeyPriv}
+	estimator := &lnwallet.StaticFeeEstimator{FeeRate: 50}
+
+	alice2, err := lnwallet.NewLightningChannel(aliceSigner, notifier,
+		estimator, alice.State())
+	if err != nil {
+		t.Fatalf("unable to create test channels: %v", err)
+	}
+	defer alice2.Stop()
+
+	// Send this newer channel to breach arbiter, which should replace the
+	// prior.
+	select {
+	case brar.newContracts <- alice2:
+	case <-time.After(500 * time.Millisecond):
+		t.Fatalf("unable to register alice with breach arbiter: %v", err)
+	}
+
+	// Signal a spend of the funding transaction and wait for the close
+	// observer to exit. This time we are allowing the handoff to succeed.
+	notifier.Spend(chanPoint, 100, breachTxn)
+	alice2.WaitForClose()
+
+	// Check that the breach was properly recorded in the breach arbiter,
+	// and that the close observer marked the channel as pending closed
+	// before exiting.
+	assertArbiterBreach(t, brar, chanPoint)
+	assertPendingClosed(t, alice)
+}
+
+// assertArbiterBreach checks that the breach arbiter has persisted the breach
+// information for a particular channel.
+func assertArbiterBreach(t *testing.T, brar *breachArbiter,
+	chanPoint *wire.OutPoint) {
+
+	isBreached, err := brar.IsBreached(chanPoint)
+	if err != nil {
+		t.Fatalf("unable to determine if channel is "+
+			"breached: %v", err)
+	}
+
+	if !isBreached {
+		t.Fatalf("channel %v was never marked breached",
+			chanPoint)
+	}
+
+}
+
+// assertNoArbiterBreach checks that the breach arbiter has not persisted the
+// breach information for a particular channel.
+func assertNoArbiterBreach(t *testing.T, brar *breachArbiter,
+	chanPoint *wire.OutPoint) {
+
+	isBreached, err := brar.IsBreached(chanPoint)
+	if err != nil {
+		t.Fatalf("unable to determine if channel is "+
+			"breached: %v", err)
+	}
+
+	if isBreached {
+		t.Fatalf("channel %v was marked breached",
+			chanPoint)
+	}
+}
+
+// assertPendingClosed checks that the channel has been marked pending closed in
+// the channel database.
+func assertPendingClosed(t *testing.T, c *lnwallet.LightningChannel) {
+	closedChans, err := c.State().Db.FetchClosedChannels(true)
+	if err != nil {
+		t.Fatalf("unable to load pending closed channels: %v", err)
+	}
+
+	for _, chanSummary := range closedChans {
+		if !chanSummary.IsPending {
```

Comment:
Doesn't seem like this check is needed, since 
1) you fetch pending only channels.
~~2) If non-pending channels are fetched, this should throw error if `!chanSummary.IsPending && chanSummary.ChanPoint == *c.ChanPoint` instead of just continuing.~~ Realized you will hit the `t.Fatal` below.

Reply:
you're right, i think this is unnecessary since we only ask for pending 👍